### PR TITLE
Add grpc-gateway v2.17.0

### PR DESCRIPTION
**Pull request by pgmitche**
Reapplies patch cleanly on 2.17.0

### CODE REVIEW

**Review by pkwarren — APPROVED** [review]
Looks good - verified code generation.

File: plugins/grpc-ecosystem/gateway/v2.17.0/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# syntax=docker/dockerfile:1.4
+FROM golang:1.21.0-bullseye AS build
```
Comment (pkwarren) [inline plugins/grpc-ecosystem/gateway/v2.17.0/Dockerfile:2]:
```suggestion
FROM golang:1.20.7-bullseye AS build
```

**Review by pkwarren — COMMENTED** [review]
For some reason 1.21.0 had an error running tests - trying reverting to latest 1.20.x.

File: plugins/grpc-ecosystem/gateway/v2.17.0/Dockerfile (hunk 1 above)
Comment (pgmitche) [reply plugins/grpc-ecosystem/gateway/v2.17.0/Dockerfile]: thanks

**Pull request merged**